File: program/find-sum-of-digits-of-a-number-by-recursion/FindSumOfDigitsOfANumberByRecursion.java
public class FindSumOfDigitsOfANumberByRecursion {
    public static void main(String[] args) {

        int num = 12345;
        System.out.print(sum(num));
    }

    static int sum(int num) {
        if (num == 0) {
            return 0;
        }

        return (num % 10 + sum(num / 10));
    }
}

File: program/find-sum-of-digits-of-a-number-by-recursion/FindSumOfDigitsOfANumberByRecursion.js
function sumOfDigits (num) {
    if(num < 10)
    {
        return num;
    }
    else
    {
        return num%10 + sumOfDigits(Math.floor(num/10));
    }
}

sumOfDigits(12345);

console.log("Sum of Digits of sumOfDigits(12345) is", sumOfDigits(12345));
